lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/LazyShapeDtype.cpp

```diff
@@ -0,0 +1,31 @@
+/**
+ * This is a handwritten file that accompanies codegenerated header
+ * LazyShapeDtype.h
+ *
+ * The purpose of these shape/dtype inference methods are to fill gaps
+ * where we do not yet have structured kernels in pytorch core.  Ops
+ * for which there _are_ structured kernels can use meta::op() to infer
+ * shape/dtype, and codegen makes use of this.  Ops for which there are not
+ * yet structured kernels can still be used with lazy_tensor codegen, but require
+ * manual intervention to implement compute_shape_{op} and compute_dtype_{op}.
+ *
+ */
+#include "lazy_tensor_core/csrc/ts_backend/LazyShapeDtype.h"
+
+namespace torch_lazy_tensors{
+namespace ir {
+namespace ops {
+
+std::vector<int64_t> compute_shape_mean(const at::Tensor & self, c10::optional<at::ScalarType> dtype) {
```

Comment:
Nit: Since the code is handwritten, maybe it's better to have `const at::Tensor& self` instead.

Reply:
i made the change, but i'm not sure it's useful unless we also fix the generator so it generates matching headers.  The reason is that it's nice to just copy-paste the generated declaration and fill in the implementation.   i'll add a todo here to consider updating the generator correspondingly

Reply:
actually I don't know if it's worth fixing the generator for this case. might be ok to just have the inconsistency in this file.

Reply:
it will probably become more relevant when we start running  `clang-format` /`clang-tidy` and would want to avoid warnings, noise in diff due to formatting changes etc.

Reply:
For the code-gen header, it probably doesn't matter.